### PR TITLE
Don't cache connection settings from config file

### PR DESCRIPTION
**Pull request by jandubois**
When creating/restoring snapshots we may fetch the connection settings multiple times, once to stop the backend, and once to start it back up. The API credentials may change if the app is restarted during that time, so cached values may no longer be valid.

This changes makes sure that GetConnectionInfo() will update settings from the file on each call, but still allow commandline options to overwrite the file settings.

### CODE REVIEW

File: src/go/rdctl/pkg/config/config.go
Comment (rak-phillip) [inline src/go/rdctl/pkg/config/config.go:102]: Would removing the first conditional be acceptable in this case? I think that it would make the code more concise

```suggestion
	settings.Host = connectionSettings.Host
	
	if settings.Host == "" {
		settings.Host = "127.0.0.1"
```

**Review by rak-phillip — APPROVED** [review]
LGTM

Tested the scenario detailed in my earlier comment [^1] on Windows 11 and the issue is resolved. I'm just waiting on the comment from this PR before we merge. 

[^1]: https://github.com/rancher-sandbox/rancher-desktop/issues/5849#issuecomment-1817001133

File: src/go/rdctl/pkg/config/config.go
Comment (jandubois) [reply src/go/rdctl/pkg/config/config.go:102]: @rak-phillip Doesn't this throw away the `host` setting from the config file? It would be either the value of `--host`, or `127.0.0.1`, but the value in the file would always be silently ignored. Why would we want to do that?

Comment (rak-phillip) [reply src/go/rdctl/pkg/config/config.go:102]: > but the value in the file would always be silently ignored. Why would we want to do that?

You're right, then it sounds like my suggestion wouldn't be acceptable 🙂

I think that the pattern of `json.Unmarshal(content, &settings)` makes it unclear to me that we are mutating `settings` and that is why I thought it would be acceptable to update `settings` without the need for the conditional. This has more to do with low frequency that I read and write Go more than anything 🙃

**Pull request merged**
